luarules/gadgets/game_disable_assist_ally_construction.lua

```diff
@@ -0,0 +1,64 @@
+function gadget:GetInfo()
+	return {
+		name    = 'Disable Assist Ally Construction',
+		desc    = 'Disable assisting allied units (e.g. labs and units/buildings under construction) when modoption is enabled',
+		author  = 'Rimilel',
+		date    = 'April 2024',
+		license = 'GNU GPL, v2 or later',
+		layer   = 0,
+		enabled = true
+	}
+end
+
+----------------------------------------------------------------
+-- Synced only
+----------------------------------------------------------------
+if not gadgetHandler:IsSyncedCode() then
+	return false
+end
+
+local function isComplete(u)
+    local _,_,_,_,buildProgress=Spring.GetUnitHealth(u)
+    if buildProgress and buildProgress>=1 then
+        return true
+    else
+        return false
+    end
+end
+
+function gadget:AllowCommand(unitID, unitDefID, unitTeam, cmdID, cmdParams, cmdOptions, cmdTag, synced)
+
+    if(not Spring.GetModOptions().disable_assist_ally_construction) then
+        return true
+    end
```

Comment:
Yeah it's s good idea to do this. Just cache whether modoptions is enabled once and check that instead of calling GetModOptions every time. Also if its disabled, just quit gadget on init right away. For example:
```
local disableAssist = Spring.GetModOptions().disable_assist_ally_construction
function gadget:Initialize()
    if not disableAssist or disableAssist ~= true then
        gadgetHandler:RemoveGadget(self)
    end
end
-- now you can check "disableAssist" later in the code if its true you do whatever you needed to do when its on, etc.
```

Reply:
you don't need to do anything fancy like that, if a gadget returns false it doesn't get loaded, just a duplicate of the is synced check you have on line 16 to 18, except instead of checking if it is not in synced space check if the modoption is not on.

Reply:
Oh, yeah, I got used to multi-choice options.

Reply:
fixed spaces and checking modoption enabled only once